datajoint/table.py

```diff
@@ -116,29 +116,55 @@ def get_select_fields(self, select_fields=None):
         """
         return '*' if select_fields is None else self.heading.project(select_fields).as_sql
 
-    def parents(self, primary=None):
+    def parents(self, primary=None, as_objects=False):
         """
         :param primary: if None, then all parents are returned. If True, then only foreign keys composed of
             primary key attributes are considered.  If False, the only foreign keys including at least one non-primary
             attribute are considered.
-        :return: dict of tables referenced with self's foreign keys
+        :param as_objects: if False (default), the output is a dict describing the foreign keys. If True, return table objects.
+        :return: dict of tables referenced with self's foreign keys  or list of table objects if as_objects=True
         """
-        return self.connection.dependencies.parents(self.full_table_name, primary)
+        parents = self.connection.dependencies.parents(self.full_table_name, primary)
+        if as_objects:
+            parents = [FreeTable(self.connection, c) for c in parents]
+        return parents
 
-    def children(self, primary=None):
+    def children(self, primary=None, as_objects=False):
         """
         :param primary: if None, then all children are returned. If True, then only foreign keys composed of
             primary key attributes are considered.  If False, the only foreign keys including at least one non-primary
             attribute are considered.
-        :return: dict of tables with foreign keys referencing self
-        """
-        return self.connection.dependencies.children(self.full_table_name, primary)
-
-    def descendants(self):
-        return self.connection.dependencies.descendants(self.full_table_name)
-
-    def ancestors(self):
-        return self.connection.dependencies.ancestors(self.full_table_name)
+        :param as_objects: if False (default), the output is a dict describing the foreign keys. If True, return table objects.
+        :return: dict of tables with foreign keys referencing self or list of table objects if as_objects=True
+        """
+        nodes = dict((next(iter(self.connection.dependencies.children(k).items())) if k.isdigit() else (k, v))
+                     for k, v in self.connection.dependencies.children(self.full_table_name, primary).items())
```

Comment:
Could you elaborate why it is necessary to check if key is a string representation of a number? Unclear what is the overall issue this PR is trying to solve (aside from allowing objects to be returned and updates to dependency loading). Would you mind updating the `CHANGELOG.md` + and PR description?

Reply:
When DataJoint creates the dependency graph, it introduces extra nodes when the foreign key attributes have different names than the primary key attributes that they reference.  This allows creating multi-graphs where any two nodes can be connected by multiple edges. The names of such nodes are composed of incremental numbers. Thus nodes composed of numbers represent "alias nodes" that do not represent a table.

Reply:
updated CHANGELOG and version to 0.12.8